src/script/common/c_content.cpp

```diff
@@ -1943,3 +1965,31 @@ HudElementStat read_hud_change(lua_State *L, HudElement *elem, void **value)
 	}
 	return stat;
 }
+
+void push_chat_message(lua_State *L, const ChatMessage &msg)
+{
+	lua_newtable(L);
+
+	std::string type = "raw";
+	enum_to_string(es_ChatMessageType, msg.type, type);
+	setstringfield(L, -1, "type", type.c_str());
+	setstringfield(L, -1, "sender", wide_to_utf8(msg.sender).c_str());
+	setstringfield(L, -1, "text", wide_to_utf8(msg.text).c_str());
+	setintfield(L, -1, "timestamp", msg.timestamp);
+
+}
+
+ChatMessage read_chat_message(lua_State *L, int index)
```

Comment:
can you instead implement readParam<ChatMessage>(L, index) instead ? else i will do it later :p

Reply:
Yes, this will permit to update all of them at once in a PR which is dedicated to that feature.